app/gps/exposureInfo.ts

```diff
@@ -0,0 +1,59 @@
+import { ExposureInfo } from '../exposureHistory';
+
+type Event = 'onGPSExposureInfoUpdated';
+
+type HandleOnExposureInfoUpdated = (data: ExposureInfo) => void;
+
+interface Subscriber {
+  eventName: Event;
+  callback: HandleOnExposureInfoUpdated;
+}
+
+interface GPSExposureInfoEventEmitter {
+  subscribers: Subscriber[];
+  emit: (event: Event, data: ExposureInfo) => void;
+  addListener: (
+    event: Event,
+    callback: HandleOnExposureInfoUpdated,
+  ) => {
+    remove: () => void;
+  };
+}
+
+const ExposureEvents: GPSExposureInfoEventEmitter = {
```

Comment:
so this might want to be cleaned up a bit.  currently remove() , removes all of the subscribers, which maybe is _okay_ given we only expect to ever have one subscriber to this emitter, but this is not clearly indicated to the reader.

What do you think about changing this up a touch to make this more clear?

something like

```typescript
interface GPSExposureInfoEventEmitter {
  subscriber: Subscriber | null;
  emit: (event: Event, data: ExposureInfo) => void;
  addListener: (
    event: Event,
    callback: HandleOnExposureInfoUpdated,
  ) => {
    remove: () => void;
  };
}

const ExposureEvents: GPSExposureInfoEventEmitter = {
  subscriber: null,
  emit: function (event: Event, data: ExposureInfo) {
      if (this.subscriber) {
        this.subscriber.callback(data);
      }
  },
  addListener: function (event: Event, callback: HandleOnExposureInfoUpdated) {
    const nextSubscriber: Subscriber = {
      eventName: event,
      callback,
    };
    this.subscriber = nextSubscriber
    return {
      remove: () => {
        this.subscribers = null;
      },
    };
  },
};
```

Reply:
Sure thing, thanks for the tip!